src/number/n_transExt.jl

```diff
@@ -0,0 +1,360 @@
+   p = libSingular.n_Div(x.ptr, y.ptr, c.ptr)
+   return c(p)
+end
+
+###############################################################################
+#
+#   GCD and LCM
+#
+###############################################################################
+
+function gcd(x::n_transExt, y::n_transExt)
+   if x == 0 && y == 0
+      return zero(parent(x))
+   end
+   par = parent(x)
+   p = libSingular.n_Gcd(x.ptr, y.ptr, par.ptr)
+   return par(p)
+end
+
+###############################################################################
+#
+#   Unsafe functions
+#
+###############################################################################
+
+function addeq!(x::n_transExt, y::n_transExt)
+   x.ptr = libSingular.n_InpAdd(x.ptr, y.ptr, parent(x).ptr)
+   return x
+end
+
+function mul!(x::n_transExt, y::n_transExt, z::n_transExt)
+   ptr = libSingular.n_Mult(y.ptr, z.ptr, parent(x).ptr)
+   libSingular.n_Delete(x.ptr, parent(x).ptr)
+   x.ptr = ptr
+   return x
+end
+
+function add!(x::n_transExt, y::n_transExt, z::n_transExt)
+   ptr = libSingular.n_Add(y.ptr, z.ptr, parent(x).ptr)
+   libSingular.n_Delete(x.ptr, parent(x).ptr)
+   x.ptr = ptr
+   return x
+end
+
+function zero!(x::n_transExt)
+   ptr = libSingular.n_Init(0, parent(x).ptr)
+   libSingular.n_Delete(x.ptr, parent(x).ptr)
+   x.ptr = ptr
+   return x
+end
+
+###############################################################################
+#
+#   Conversions and promotions
+#
+###############################################################################
+
+promote_rule(C::Type{n_transExt}, ::Type{T}) where T <: Integer = n_transExt
+
+promote_rule(C::Type{n_transExt}, ::Type{n_Z}) = n_transExt
+
+###############################################################################
+#
+#   Parent call functions
+#
+###############################################################################
+
+function (R::N_FField)()
+   z = n_transExt(R)
+   z.parent = R
+   return z
+end
+
+function (R::N_FField)(x::Integer)
+   z = R(libSingular.n_InitMPZ(BigInt(x), R.ptr)) 
+   z.parent = R
+   return z
+end
+
+function (R::N_FField)(n::Int)
+   z = n_transExt(R, n)
+   z.parent = R
+   return z
+end
+
+(R::N_FField)(n::n_transExt) = n
+
+function (R::N_FField)(n::libSingular.number)
+   z = n_transExt(R, n) 
+   z.parent = R
+   return z
+end
+
+function (R::N_FField)(x::Nemo.fmpz)
+   a = BigInt()
+   ccall((:flint_mpz_init_set_readonly, :libflint), Nothing,
+         (Ptr{BigInt}, Ptr{fmpz}), Ref(a), Ref(x))
+   z = R(libSingular.n_InitMPZ(a, R.ptr))
+      z.parent = R
+   return z
+end
+
+###############################################################################
+#
+#   FunctionField constructor
+#
+###############################################################################
+
+@doc Markdown.doc"""
+    FunctionField(F::Field, S::Array{String, 1})
+> Returns a tuple $K, a$ consisting of a function field $K$ over the field $F$
+> with transcendence basis stored in the array $S$.
```

Comment:
What fields are supported here? And do we raise an exception if someone supplies some other field?

Reply:
I changed Fiield to Singular.Field, which should do the trick

The error returned by Julia should be sufficient.